### PR TITLE
Apply to jobs

### PR DESCRIPTION
**Pull request by arsidada**

### CODE REVIEW

File: src/Components/Dashboard/Cards/JobApplicationCard.js [hunk 1]
```diff
@@ -250,6 +252,52 @@ class JobApplicationCard extends Component {
...
+        alreadyApplied = true;
+        break;
+      }
+    }
+
+    if (!alreadyApplied) {
+      let jobAppObj = {
+        job_id: this.props.data.job_id.toString(),
+        resumes: this.getResumeURL(),
+        cover_letters: "coverletters2",
```
Comment (ahmedhamodi) [inline src/Components/Dashboard/Cards/JobApplicationCard.js:282]: Should we leave a "TODO" comment here so we know to later implement cover letter?

File: src/Components/Dashboard/Cards/JobApplicationSelfCard.js [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+  };
+
+  getResumeURL = () => {
+    const userProfile = JSON.parse(localStorage.getItem("userProfile"));
+    return userProfile["custom:resume"];
+  };
+
+  applyJob = async () => {
+    console.log(this.props.data);
+    console.log(this.getResumeURL());
```
Comment (ahmedhamodi) [inline src/Components/Dashboard/Cards/JobApplicationSelfCard.js:261]: Console logs!

File: src/Components/Dashboard/Cards/JobApplicationSelfCard.js [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+
+    let jobAppObj = {
+      job_id: this.props.data.job_id.toString(),
+      resumes: this.getResumeURL(),
+      cover_letters: "coverletters2",
```
Comment (ahmedhamodi) [inline src/Components/Dashboard/Cards/JobApplicationSelfCard.js:266]: Comment to indicate future work here? (also, is there a ticket associated with cover letter work?)

File: src/Components/Dashboard/Cards/JobApplicationSelfCard.js [hunk 4]
```diff
@@ -0,0 +1,629 @@
...
+    );
+    console.log(appResponse);
+    if (appResponse.status === 200) {
+      window.alert("Successfully applied for this job");
+      this.setState({
+        open: false,
+      });
+    } else {
+      console.log(appResponse);
+      window.alert("Failed to apply for this job!");
```
Comment (ahmedhamodi) [inline src/Components/Dashboard/Cards/JobApplicationSelfCard.js:282]: Couple more logs to remove. In the "else" clause, can do a .err!

File: src/Components/Dashboard/UserProfile.js [hunk 5]
```diff
@@ -463,6 +469,8 @@ class Landing extends Component {
     jobsDataObj.poster_family_name = userProfile.family_name;
     jobsDataObj.poster_given_name = userProfile.given_name;
 
+    console.log(jobsDataObj);
+
```
Comment (ahmedhamodi) [inline src/Components/Dashboard/UserProfile.js:473]: Console log

**Review by ahmedhamodi — APPROVED** [review]
Overall lgtm. A few comments that I would like addressed before merging!

File: src/Components/Dashboard/Cards/JobApplicationCard.js (hunk 1 above)
Comment (arsidada) [reply src/Components/Dashboard/Cards/JobApplicationCard.js:282]: no. coverletter is just a requirement on the API payload. just added it here to get it working

File: src/Components/Dashboard/Cards/JobApplicationSelfCard.js (hunk 3 above)
Comment (arsidada) [reply src/Components/Dashboard/Cards/JobApplicationSelfCard.js:266]: same as before

File: src/Components/Dashboard/Cards/JobApplicationSelfCard.js (hunk 4 above)
Comment (arsidada) [reply src/Components/Dashboard/Cards/JobApplicationSelfCard.js:282]: will make this nicer with notifications

**Pull request merged**